app/browser/api/ledger.js

```diff
@@ -706,14 +706,16 @@ const excludeP = (publisherKey, callback) => {
   })
 }
 
-const addSiteVisit = (state, timestamp, location, tabId) => {
+const addSiteVisit = (state, timestamp, location, tabId, manualAdd = false) => {
   if (!synopsis) {
     return state
   }
 
-  location = pageDataUtil.getInfoKey(location)
-  const locationData = ledgerState.getLocation(state, location)
-  const duration = new Date().getTime() - timestamp
+  location = manualAdd ? tldjs.getDomain(location) : pageDataUtil.getInfoKey(location)
```

Comment:
Please add check if `tldjs.getDomain(location)` is null

Reply:
why can't this just use `pageDataUtil.getInfoKey`? I'm slightly concerned that pageDataUtil.getInfoKey includes the URL protocol whereas tldjs.getDomain doesn't

Reply:
So here lays the problem I'm having and the way I chose to test it revealed this:

In the case we are adding something via the history item context menu, that site has already been visited and in some capacity captured in ledger state. So using getInfoKey() for ledgerState.getLocation(state, location) pulls in location data just fine.

The same will work for bookmarks added with the '+' (you would have had to visit the page already)

WHere tldjs.getDomain becomes helpful is if I make a bookmark for a site that I haven't visited yet and manually enter the location. In that case ledgerState.getLocation(state, location) returns nothing as it has no record of the site. This way a key for the domain can be generated without depending on getLocation()

With this in mind, I would be compelled to make the use of tldjs.getDomain a special case only for those bookmarks, not history entries since we can trust the ledgerState for the location data. Since the publisher key ends up being the url without the protocol, it doesn't seem to have any unintended side effects.

Let me know your thoughts :) @NejcZdovc @diracdeltas

Reply:
@RyanML not sure i understand the problem since this isn't calling `ledgerState.getLocation(state, location)`. i'm just suggesting changing this line to `location = pageDataUtil.getInfoKey(location)` instead of `location = manualAdd ? tldjs.getDomain(location) : pageDataUtil.getInfoKey(location)`

Reply:
your comment would make sense for https://github.com/brave/browser-laptop/pull/13856/files#diff-3cdfdb124f718787d584e9e2b82fcedcR720

Reply:
@diracdeltas per our conversation, tldjs.getDomain() is no longer in use. Additionally, the context menu item is disabled for custom bookmark locations that are either invalid or without a protocol.